src/main/java/org/dave/compactmachines3/CompactMachines3.java

```diff
@@ -7,39 +7,54 @@
+import net.minecraftforge.fml.common.event.FMLServerAboutToStartEvent;
+import net.minecraftforge.fml.common.event.FMLServerStartingEvent;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.dave.compactmachines3.command.CommandCompactMachines3;
 import org.dave.compactmachines3.gui.GuiHandler;
 import org.dave.compactmachines3.integration.CapabilityNullHandlerRegistry;
 import org.dave.compactmachines3.miniaturization.MultiblockRecipes;
-import org.dave.compactmachines3.misc.*;
+import org.dave.compactmachines3.misc.ConfigurationHandler;
+import org.dave.compactmachines3.misc.CreativeTabCompactMachines3;
+import org.dave.compactmachines3.misc.MachineEventHandler;
+import org.dave.compactmachines3.misc.PlayerEventHandler;
+import org.dave.compactmachines3.misc.RenderTickCounter;
 import org.dave.compactmachines3.network.PackageHandler;
 import org.dave.compactmachines3.proxy.CommonProxy;
+import org.dave.compactmachines3.reference.EnumMachineSize;
 import org.dave.compactmachines3.render.BakeryHandler;
 import org.dave.compactmachines3.schema.SchemaRegistry;
 import org.dave.compactmachines3.skyworld.SkyChunkGenerator;
 import org.dave.compactmachines3.skyworld.SkyDimension;
 import org.dave.compactmachines3.skyworld.SkyWorldEvents;
 import org.dave.compactmachines3.skyworld.SkyWorldSavedData;
 import org.dave.compactmachines3.utility.AnnotatedInstanceUtil;
-import org.dave.compactmachines3.utility.Logz;
 import org.dave.compactmachines3.world.ChunkLoadingMachines;
 import org.dave.compactmachines3.world.ClientWorldData;
 import org.dave.compactmachines3.world.WorldSavedDataMachines;
 import org.dave.compactmachines3.world.data.provider.ExtraTileDataProviderRegistry;
 import org.dave.compactmachines3.world.tools.DimensionTools;
 
-@Mod(modid = CompactMachines3.MODID, version = CompactMachines3.VERSION, acceptedMinecraftVersions = "[1.12,1.13)", dependencies = "after:refinedstorage;after:yunomakegoodmap", guiFactory = CompactMachines3.GUI_FACTORY)
-public class CompactMachines3
-{
+import java.util.Map;
+
+@Mod(modid = CompactMachines3.MODID, version = CompactMachines3.VERSION, acceptedMinecraftVersions = "[1.12,1.13)",
+        dependencies = "after:refinedstorage;after:yunomakegoodmap", guiFactory = CompactMachines3.GUI_FACTORY)
+public class CompactMachines3 {
     public static final String MODID = "compactmachines3";
-    public static final String VERSION = "3.0.18";
+    public static final String VERSION = "@VERSION@";
     public static final String GUI_FACTORY = "org.dave.compactmachines3.misc.ConfigGuiFactory";
 
     @Mod.Instance(CompactMachines3.MODID)
     public static CompactMachines3 instance;
+    public static final Logger logger = LogManager.getLogger(CompactMachines3.MODID);
 
     public static ClientWorldData clientWorldData;
+    public static Map<Integer, BlockPos> clientMachineGrid;
+    public static Map<Integer, EnumMachineSize> clientMachineSizes;
```

Comment:
This seems a bit odd placed here; done in a WSD instance on the 1.16 version. Any particular reasoning?

Reply:
No particular reason; I mainly did it since there is another client-side field that used the main mod class for this even though it is quite improper. I can move this to the WSD class once I get home. Collaborators also have access to push commits to my fork if you care enough to do it yourself.

Reply:
Fixed